### PR TITLE
separating keys from firmware

### PR DESCRIPTION
**Pull request by amundas**
Adding this PR quite early to show what I had in mind for the post.

### CODE REVIEW

**Comment by tyhoff** [comment]
Getting to this now. It sounds like @franc0is is going to hopefully have a post ready for tomorrow, so we'll follow this up after that. I'm going to push a commit or two to this for things that I don't want to burden you with (formatting, header/footer, etc).

**Review by tyhoff — COMMENTED** [review]
Looking good thus far. I think the last true thing I need from you is to fill out the information you'd like to share on Interrupt here: https://github.com/memfault/interrupt/blob/master/_data/authors.yml. I didn't want put items there that you weren't comfortable sharing.

**Comment by tyhoff** [comment]
Did you have other changes you wanted to make to this post?

File: _drafts/separating-keys-from-firmware.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+that we can use to get the parameters and register new devices. The firmware for
+these example devices is written in C. The environment in this example uses
+bash, GCC, a makefile, and J-Link Commander for programming.
+
+### A note on security
+
+In this example, we will store an encryption key in standard flash memory.
+Whether this is a good idea or not depends on your application, although most
+MCUs allow you to enable readback protection, these mechanisms have been
+defeated for a number of popular devices. Secure key storage can be done with
```
[the remark this answers is not in the document]
Comment (amundas) [reply _drafts/separating-keys-from-firmware.md]: Added a couple of references for that

**Comment by amundas** [comment]
> Did you have other changes you wanted to make to this post?

I had some rephrasings I wanted to make, but they were all covered by your changes

**Pull request merged**